### PR TITLE
Experimentation with App Insights (Intergen)

### PR DESCRIPTION
**Pull request by steven2000nz**

### CODE REVIEW

File: docs/HOL-Usage_Analysis_With_Application_Insights/README.md [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+====================================================================================
+Users are reporting slow response times on the PartsUnlimited website. None of our current telemetry is logging custom events. We want to track what items are being removed from the cart and how long it is taking from the site. We also want to dive a bit deeper in to our current analytic reports and dig up some useful data to provide a better experience for our users.
+
+### Pre-requisites: ###
+- Visual Studio 2015 or higher
+
+- Application Insights created in the same Azure Resource Group as the PartsUnlimited website
+
+### Tasks Overview: ###
+**Task 1. Set up Application Insights for PartsUnlimited**
```
Comment (nzthiago) [inline docs/HOL-Usage_Analysis_With_Application_Insights/README.md]: Each needs a description

**Comment by wbuchwalter** [comment]
Same comment as in #89, could we remove the first part (setup of App Insights) and instead link [to the official doc](https://github.com/Microsoft/ApplicationInsights-aspnetcore/wiki/Getting-Started)?

**Comment by wbuchwalter** [comment]
When embedding code into the documentation please use [the correct GitHub markdown syntax](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#code), this will bring highlight syntaxing.

**Comment by julienstroheker** [comment]
@wbuchwalter please review and merge if you are ok.

**Comment by wbuchwalter** [comment]
Syntax highlighting is still not correct. Please use: 
* ```C# for c# sample
* ```Javascript for JS samples etc...

For example:
* Without highlighting:
```
public class ApplicationInsightsTelemetryProvider : ITelemetryProvider
    {
        private readonly TelemetryClient _telemetry;

        public ApplicationInsightsTelemetryProvider()
        {
            _telemetry = new TelemetryClient();
        }
}
```

* With highlighting:
```C#
public class ApplicationInsightsTelemetryProvider : ITelemetryProvider
    {
        private readonly TelemetryClient _telemetry;

        public ApplicationInsightsTelemetryProvider()
        {
            _telemetry = new TelemetryClient();
        }
}
```

**Comment by wbuchwalter** [comment]
Thanks!

**Pull request merged**